examples/messages-forall.cpp

```diff
@@ -0,0 +1,461 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-25, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+
+#include "memoryManager.hpp"
+
+#include "RAJA/RAJA.hpp"
+#include "RAJA/util/resource.hpp"
+
+/*
+ *  Vector Addition Example
+ *
+ *  Computes c = a + b, where a, b, c are vectors of ints.
+ *  It illustrates similarities between a  C-style for-loop and a RAJA 
+ *  forall loop with using RAJA messages to print out if a value is 
+ *  negative.
+ *
+ *  Note: this example is the same as resource-forall.cpp with
+ *  some additional logger to store how the message handler can
+ *  be used as a basic logger for both host and device. 
+ *
+ *  RAJA features shown:
+ *    - `forall` loop iteration with messages
+ *    -  Create message handler with Resource argument
+ *
+ */
+
+// This is a simplified example fixed string to show how
+// custom types can be used with the message queue.
+template <std::size_t N>
+class my_string
```

Comment:
Thanks for adding an example with a "string-like" type in it :) It's helps provide me a road map for how I could replace my current Warning / Error reporting on the GPUs with this framework.